### ISSUE TITLE
Messageable.send sometimes raises InvalidArgument for MessageReference `reference` argument

### ISSUE THREAD

**Issue by zephyrkul** [Rapptz/discord.py#6726]
### Summary

Pre-existing MessageReference objects don't have `fail_if_not_exists` as an attribute, causing an AttributeError in `to_dict()`

### Reproduction Steps

1) Get a pre-existing MessageReference object, e.g. from `Message.reference`
2) Attempt to use said reference in `Messageable.send()`

### Minimal Reproducible Code

```python
@bot.listen("on_message")
async def on_message(message):
  if message.reference:
    # raises InvalidArgument
    await message.channel.send("Ping!", reference=message.reference)
    # raises AttributeError
    ref_dict = message.reference.to_message_reference_dict()
```

### Expected Results

For the bot to reply to the message that the received message replied to.

### Actual Results

```py
Traceback (most recent call last):
  File "/home/zephyrkul/venv/lib/python3.8/site-packages/discord/abc.py", line 1034, in send
    raise InvalidArgument('reference parameter must be Message or MessageReference') from None
discord.errors.InvalidArgument: reference parameter must be Message or MessageReference
```
(And this is the error that's suppressed by `from None` above)
```py
Traceback (most recent call last):
  File "/home/zephyrkul/venv/lib/python3.8/site-packages/discord/message.py", line 396, in to_dict
    if self.fail_if_not_exists is not None:
AttributeError: fail_if_not_exists
```

### Intents

`discord.Intents.all()`

### System Information

- Python v3.8.5-final
- discord.py v1.7.1-final
- aiohttp v3.7.3
- system info: Linux 5.4.0-71-generic #79-Ubuntu SMP Wed Mar 24 10:56:57 UTC 2021

### Checklist

I have searched the open issues for duplicates.
I have shown the entire traceback, if possible.
I have removed my token from display, if visible.

### Additional Context

Right now this issue can be bypassed by using `Message.reference.resolved` rather than `Message.reference`.

**Comment by TK421bsod** [comment]
This appears to have been fixed by commit da6119e.

**Comment by zephyrkul** [comment]
So it seems, thanks. I was only checking the definition of `to_dict()` to see if this bug was fixed, didn't think to check `__init__` too.
